heron/stmgr/src/cpp/manager/stateful-restorer.cpp

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright 2015 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "manager/stateful-restorer.h"
+#include <functional>
+#include <iostream>
+#include <list>
+#include <map>
+#include <set>
+#include <string>
+#include <vector>
+#include "manager/stmgr-server.h"
+#include "manager/ckptmgr-client.h"
+#include "manager/stmgr-clientmgr.h"
+#include "util/tuple-cache.h"
+#include "metrics/metrics.h"
+#include "proto/messages.h"
+#include "basics/basics.h"
+#include "errors/errors.h"
+#include "threads/threads.h"
+#include "network/network.h"
+
+namespace heron {
+namespace stmgr {
+
+// Stats for restore
+const sp_string METRIC_START_RESTORE = "__start_restore";
+const sp_string METRIC_START_RESTORE_IN_PROGRESS = "__start_restore_in_progress";
+const sp_string METRIC_START_RESTORE_FAILED = "__start_restore_failed";
+const sp_string METRIC_CKPT_REQUESTS = "__ckpt_requests";
+const sp_string METRIC_CKPT_RESPONSES = "__ckpt_responses";
+const sp_string METRIC_CKPT_RESPONSES_IGNORED = "__ckpt_responses_ignored";
+const sp_string METRIC_CKPT_RESPONSES_ERROR = "__ckpt_responses_error";
+const sp_string METRIC_INSTANCE_RESTORE_REQUESTS = "__instance_restore_requests";
+const sp_string METRIC_INSTANCE_RESTORE_RESPONSES = "__instance_restore_responses";
+const sp_string METRIC_INSTANCE_RESTORE_RESPONSES_IGNORED = "__instance_restore_response_ignored";
+
+StatefulRestorer::StatefulRestorer(CkptMgrClient* _ckptmgr,
```

Comment:
can you clarify the responsibilities of the restorer in comments. What does it do or manage and why/when/by whom is it called?